src/features/chart/components/Chart.js

```diff
@@ -26,65 +26,66 @@ const tenors = [
 ];
 
 export const Chart = () => {
-  const [chartRange, setChartRange] = useState("5y");
   const dispatch = useDispatch();
-  const { selectedChartData } = useSelector(state => state.chartData);
+  const { selectedChartData, selectedChartRange } = useSelector(
+    state => state.chartData
+  );
   const { selectedStockTicker } = useSelector(state => state.stockTickerData);
   const updateChartRange = stock => dispatch(updateChartAction(stock));
   const onClickHandler = event => {
     updateChartRange(event.target.value);
-    setChartRange(event.target.value);
   };
 
-  const formatChartData = () => {
-    const formatDate = isoDate => {
-      const date = new Date(isoDate);
-      switch (chartRange) {
-        case "max":
-          return Intl.DateTimeFormat("en-US", {
-            year: "2-digit",
-            month: "short"
-          }).format(date);
-        case "5y":
-          return Intl.DateTimeFormat("en-US", {
-            year: "2-digit",
-            month: "short"
-          }).format(date);
-        case "1y":
-          return Intl.DateTimeFormat("en-US", { month: "short" }).format(date);
-        case "1m":
-          return Intl.DateTimeFormat("en-US", {
-            year: "2-digit",
-            month: "short"
-          }).format(date);
-        case "5d":
-          return Intl.DateTimeFormat("en-US", { weekday: "short" }).format(
-            date
-          );
-        case "1d":
-          return Intl.DateTimeFormat("en-US", {
-            hour: "2-digit",
-            minute: "2-digit"
-          }).format(date);
-        default:
-          return isoDate;
-      }
-    };
+  const formatDate = isoDate => {
```

Comment:
Can this be taken out of the component and made into a pure function?

Reply:
But our switch condition is selectedChartRange, which need to be inside the component

Reply:
You can change the function arguments to include that as a 2nd argument:
```
const formatDate = (isoDate, selectedChartRange) => {
```